api/src/main/java/net/kyori/adventure/text/format/StyleImpl.java

```diff
@@ -117,56 +105,29 @@ static void decorate(final Builder builder, final TextDecoration[] decorations)
 
   @Override
   public TextDecoration.@NotNull State decoration(final @NotNull TextDecoration decoration) {
-    if (decoration == TextDecoration.BOLD) {
-      return this.bold;
-    } else if (decoration == TextDecoration.ITALIC) {
-      return this.italic;
-    } else if (decoration == TextDecoration.UNDERLINED) {
-      return this.underlined;
-    } else if (decoration == TextDecoration.STRIKETHROUGH) {
-      return this.strikethrough;
-    } else if (decoration == TextDecoration.OBFUSCATED) {
-      return this.obfuscated;
+    final @Nullable TextDecoration.State state = this.decorations.get(decoration);
+    if (state != null) {
+      return state;
     }
     throw new IllegalArgumentException(String.format("unknown decoration '%s'", decoration));
   }
 
   @Override
   public @NotNull Style decoration(final @NotNull TextDecoration decoration, final TextDecoration.@NotNull State state) {
     requireNonNull(state, "state");
-    if (decoration == TextDecoration.BOLD) {
-      return new StyleImpl(this.font, this.color, this.obfuscated, state, this.strikethrough, this.underlined, this.italic, this.clickEvent, this.hoverEvent, this.insertion);
-    } else if (decoration == TextDecoration.ITALIC) {
-      return new StyleImpl(this.font, this.color, this.obfuscated, this.bold, this.strikethrough, this.underlined, state, this.clickEvent, this.hoverEvent, this.insertion);
-    } else if (decoration == TextDecoration.UNDERLINED) {
-      return new StyleImpl(this.font, this.color, this.obfuscated, this.bold, this.strikethrough, state, this.italic, this.clickEvent, this.hoverEvent, this.insertion);
-    } else if (decoration == TextDecoration.STRIKETHROUGH) {
-      return new StyleImpl(this.font, this.color, this.obfuscated, this.bold, state, this.underlined, this.italic, this.clickEvent, this.hoverEvent, this.insertion);
-    } else if (decoration == TextDecoration.OBFUSCATED) {
-      return new StyleImpl(this.font, this.color, state, this.bold, this.strikethrough, this.underlined, this.italic, this.clickEvent, this.hoverEvent, this.insertion);
-    }
-    throw new IllegalArgumentException(String.format("unknown decoration '%s'", decoration));
+    final @NotNull TextDecoration.State thisState = this.decoration(decoration);
```

Comment:
Minor nitpick, but this can just be inlined into the if as it's not used elsewhere.

Reply:
e285a7ec967d0890c07a0e015a83dbb0e408124d